barretenberg/cpp/src/barretenberg/honk/transcript/transcript.hpp

```diff
@@ -85,17 +85,17 @@ template <typename FF> class BaseTranscript {
      */
     [[nodiscard]] std::array<uint8_t, HASH_OUTPUT_SIZE> get_next_challenge_buffer() const
     {
+        // Empty buffer to compare against
+        std::array<uint8_t, HASH_OUTPUT_SIZE> empty_challenge_buffer{};
         // Prevent challenge generation if nothing was sent by the prover.
-        ASSERT(!current_round_data.empty());
+        ASSERT(!(current_round_data.empty() && previous_challenge_buffer == empty_challenge_buffer));
 
         // concatenate the hash of the previous round (if not the first round) with the current round data.
         // TODO(Adrian): Do we want to use a domain separator as the initial challenge buffer?
         // We could be cheeky and use the hash of the manifest as domain separator, which would prevent us from having
         // to domain separate all the data. (See https://safe-hash.dev)
         std::vector<uint8_t> full_buffer;
-        if (round_number > 0) {
-            full_buffer.insert(full_buffer.end(), previous_challenge_buffer.begin(), previous_challenge_buffer.end());
-        }
+        full_buffer.insert(full_buffer.end(), previous_challenge_buffer.begin(), previous_challenge_buffer.end());
```

Comment:
Should be able to condition on previous_challenge_buffer != empty_challenge_buffer to avoid appending zeros. But actually I'm confused as to why the ASSERT a few lines above does not trigger for the first round..

Reply:
the assert doesn't trigger because current_round_data isn't empty the first time we call this function